compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -1666,9 +1656,11 @@ void createTrapExpr(DiagnosticPos pos, Set<Whitespace> ws) {
         addExpressionNode(trapExpr);
     }
 
-    void endFunctionDef(DiagnosticPos pos, Set<Whitespace> ws, boolean publicFunc, boolean remoteFunc,
-                        boolean nativeFunc, boolean privateFunc, boolean bodyExists, boolean isLambda) {
+    void endFunctionDefinition(DiagnosticPos pos, Set<Whitespace> ws, String funcName, DiagnosticPos funcNamePos,
+                               boolean publicFunc, boolean remoteFunc, boolean nativeFunc, boolean privateFunc,
+                               boolean bodyExists, boolean isLambda) {
```

Comment:
From what i know requiring this many parameters in a function can be considered as a bad practice. IMO it is better if we can think of an alternative approach.

WDYT?

Reply:
Maybe not for the immediate release, but in the long term :) 
I feel like there is an object hiding there (FunctionType?) instead of those booleans.

Reply:
Yeah, I agree. It doesn't look right. Each time some new property gets added, you most likely have to add a new param to the relevant function the package builder as well. Since we'll be moving to a new parser, not sure whether this package builder will be relevant when that happens. So don't see much point in putting a lot of effort to refactor this.